### ISSUE TITLE
Revisit Maven configuration on the umbrella bundle so that building the sources with the the various bundles becomes an easier endeavour

### ISSUE THREAD

**Issue by tarelli** [openworm/org.geppetto#494]

**Comment by borismarin** [comment]
https://git-scm.com/docs/git-submodule

**Comment by LordKrabo** [comment]
Could Otto be a good alternative? https://www.ottoproject.io/intro/vagrant-successor.html

**Comment by borismarin** [comment]
@LordKrabo that would apply to #493, wouldn't it?

**Comment by LordKrabo** [comment]
Sorry, my bad.

**Comment by tarelli** [comment]
I played a bit with this today. If we configure maven to leverage more the parent bundle (putting there common configurations and having the submodules pointing to it) I got to a stage where building of individual submodules doesn't work anymore if they depend on siblings modules which would oblige us to always do mvn install for the parent module (which works). Having to always have to reinstall everything might be annoying.
